research/07-realisation.tex

```diff
@@ -1,14 +1,54 @@
 \section{Realisation of the Product}
 \label{sec:realisation-product}
 
-\subsection{Methodologies used during the project}
-\label{ssec:methodologies}
+The previous section analysed the requirements for the product as set in
+\cref{sec:requirement-analysis}. The purpose of this section is to discuss the
+frameworks and tools used during the development of the product.
+
+\subsection{Development frameworks}
+\label{ssec:frameworks}
+
+Spoofax is an already existing project. In order for the product to be as easily
+integrated as possible, most of the tooling used by Spoofax will be reused. This
+means that the Java programming language is to be used, together with Maven for
+the build environment, JUnit for unit tests and TravisCI for continuous
+integration. On top of this, Spoofax uses a few open source Java libraries as
+can be seen in the list below. It is most likely that these will be used in the
+product as well.
+
+\begin{itemize}
+  \item Guice, a lightweight dependency injection framework;
+  \item Guava, provides general utilities missing in the JDK;
+  \item RxJava, a library for composing asynchronous and event-based programs
+    using observable sequences;
+  \item Apache VFS, a library for accessing various different file systems;
```

Comment:
What about saving and restoring the state of REPLs? Likely we'll write this to a file in the filesystem somewhere.